bson/src/main/scala/bson.scala

```diff
@@ -17,6 +17,7 @@ package reactivemongo.bson
 
 object `package` extends DefaultBSONHandlers {
   type BSONElement = (String, BSONValue)
+  type BSONCodec[A] = BSONDocumentReader[A] with BSONDocumentWriter[A] with BSONHandler[BSONDocument, A]
```

Comment:
@sgodbillon agree! I just used that name because I saw it other lib, but indeed what you suggest is more consistent with `ReactiveMongo`